vivisect/base.py

```diff
@@ -684,13 +684,13 @@ def _doSetEndian(self, endian):
 #  setMeta key callbacks
 #
     def _mcb_Architecture(self, name, value):
-        # This is for legacy stuff...
-        self.arch = envi.getArchModule(value)
-        self.psize = self.arch.getPointerSize()
-
         archid = envi.getArchByName(value)
         self.setMemArchitecture(archid)
 
+        # This is for legacy stuff...
+        self.arch = self.imem_archs[envi.ARCH_DEFAULT]
+        self.psize = self.arch.getPointerSize()
+
```

Comment:
Reordering these was necessary to let the PowerPC VLE meta stuff work. Without this move the ARCH_DEFAULT emulator wouldn't be reconfigured to be the ArchModule defined in the `setMemArchitecture` call, which means that when `self.arch` is defined it would get whatever the previous default emulator was and the switch to PowerPC would come afterwards.

It took a little tracking down but this was necessary to deal with some strange VLE meta inconsistencies I saw during testing.